### DISCUSSION TITLE
Texture Map Defragmentation increases the size of the texture

### DISCUSSION THREAD

**Discussion by shadows44** [cnr-isti-vclab/meshlab#1003] [Q&A]
Hello, first of all thanks for all the work being done on MeshLab !

I've been testing the new "Texture Map Defragmentation" on a few 3D scans (mainly created using Metashape). The fragmentation issue is indeed vastly improved, however I keep getting another issue : the output texture is larger than the input texture. For example, if I process a 3D model with a fragmented 4k texture, the output will be a 3D model with two textures (one 4K texture, one 2048x672 pixels texture).

The paper "Texture Defragmentation for Photo-Reconstructed 3D Models" does explain that this can sometimes happen, but this is presented as a rare issue. I processed 5 different 3D models, and I had this issue every single time.

Do you know how to solve this ? Maybe by tweaking the default settings ?

**Comment by alemuntoni** [comment]
Maybe @maggio-a can help on answering this question!

**Reply by cignoni** [reply]
Hi, 
thanks for testing it! A bit of fine tuning is always needed when releasing research code in the wild. 
As you can imagine having your examples to test and to check it would be of paramount importance to make things better... 
We understand privacy and confidentiality issues for 3D models, so the best thing is that you send me a mail (paolo.cignoni@isti.cnr.it) with a link to the model and we will investigate on the issue.

Thanks! 
Paolo

**Reply by shadows44** [reply]
Thanks, @cignoni ! I just sent you an e-mail with 4 models that have this issue.
